application/src/main/kotlin/com/fortysevendegrees/blogpost/db/algebra/TechnologyAlgebra.kt

```diff
@@ -1,33 +1,76 @@
 package com.fortysevendegrees.blogpost.db.algebra
 
+import arrow.core.computations.nullable
 import com.fortysevendegrees.blogpost.db.BlogpostDb
 import com.fortysevendegrees.blogpost.db.entity.TechnologyEntity
 import com.fortysevendegrees.blogpost.db.model.PostBodyTechnology
 import com.fortysevendegrees.blogpost.db.model.Technology
 import com.fortysevendegrees.blogpost.db.model.toDomain
+import com.fortysevendegrees.blogpost.db.utils.query
+import com.squareup.sqldelight.ColumnAdapter
+import com.squareup.sqldelight.Query
+import com.squareup.sqldelight.db.SqlDriver
+import com.squareup.sqldelight.db.SqlPreparedStatement
 
 interface TechnologyAlgebra {
 
-  suspend fun save(element: PostBodyTechnology): Unit
+  suspend fun save(element: PostBodyTechnology): Technology?
 
   suspend fun findByNameOrNull(name: String): Technology?
 
   suspend fun findAll(): List<Technology>
 
   companion object {
-    operator fun invoke(database: BlogpostDb) = object : TechnologyAlgebra {
-      override suspend fun save(element: PostBodyTechnology): Unit =
-        database.technologyEntityQueries.insert(
-          name = element.name,
-          description = element.description,
-          keywords = element.keywords
-        )
+    operator fun invoke(
```

Comment:
Nit: Can we rename this to default or something indicating this is an Algebra instance?
Unless `invoke` is descriptive enough for readers.